### PR TITLE
Rubocop added + violation fixes

### PR DESCRIPTION
**Pull request by ralfclaassens**

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 
-desc "Run specs"
+desc 'Run specs'
 RSpec::Core::RakeTask.new do |t|
   t.verbose = false
   t.rspec_opts = '--color --order random'
 end
 
-task :default => :spec
+task default: :spec
```
Comment (ggilder) [inline Rakefile]: This will break the current support for 1.8.7... I'm not against dropping support for 1.8.7, just noting it. I'll go ahead and bump the major version after this is merged.

File: bin/commander [hunk 2]
```diff
@@ -10,28 +10,28 @@ program :description, 'Commander utility program.'
...
-  c.action do |args, options|
+  c.action do |args, _options|
     file = args.shift || abort('file argument required.')
     name = ask 'Machine name of program: '
     description = ask 'Describe your program: '
     commands = ask_for_array 'List the commands you wish to create: '
     begin
-      File.open(file, 'w') do |f|
-        f.write <<-"...".gsub!(/^ {10}/, '')
+       File.open(file, 'w') do |f|
```
Comment (ggilder) [inline bin/commander]: What's happening here? Seems like the indentation was correct before

File: bin/commander [hunk 3]
```diff
@@ -10,28 +10,28 @@ program :description, 'Commander utility program.'
...
-          
+
           program :version, '0.0.1'
           program :description, '#{description}'
-           
+
         ...
-        commands.each do |command|
-          f.write <<-"...".gsub!(/^ {12}/, '')
+         commands.each do |command|
```
Comment (ggilder) [inline bin/commander]: same odd indentation

File: bin/commander [hunk 4]
```diff
@@ -44,12 +44,12 @@ command :init do |c|
             end
 
           ...
-        end
+         end
```
Comment (ggilder) [inline bin/commander]: Seems off, the following 6 lines actually need to be outdented one space

Comment (ggilder) [inline bin/commander]: Could be a bug in Rubocop, maybe worth adding a test case to Rubocop

File: lib/commander/command.rb [hunk 5]
```diff
@@ -3,45 +3,44 @@
...
       end
-      
+
       def __hash__
         @table
       end
-      
-      def method_missing meth, *args, &block
+
+      def method_missing(meth, *args, &_block)
```
Comment (ggilder) [inline lib/commander/command.rb]: You could actually just leave off the block arg

File: lib/commander/command.rb [hunk 6]
```diff
@@ -105,88 +104,88 @@ def example description, command
...
+    def call(args = [])
       object = @when_called.shift
       meth = @when_called.shift || :call
       options = proxy_option_struct
       case object
-      when Proc  ; object.call(args, options)
-      when Class ; meth != :call ? object.new.send(meth, args, options) : object.new(args, options)
+      when Proc then object.call(args, options)
+      when Class then meth != :call ? object.new.send(meth, args, options) : object.new(args, options)
       else         object.send(meth, args, options) if object
```
Comment (ggilder) [inline lib/commander/command.rb]: Let's go ahead and remove the extra whitespace after `else` since these no longer line up at all

File: lib/commander/user_interaction.rb [hunk 7]
```diff
@@ -164,31 +162,30 @@ def speak message, voice = :Alex, rate = 175
...
+
+    def converse(prompt, responses = {})
+      i, commands = 0, responses.map { |_key, value| value.inspect }.join(',')
       statement = responses.inject '' do |statement, (key, value)|
-        statement << (((i += 1) == 1 ? 
-          %(if response is "#{value}" then\n):
-            %(else if response is "#{value}" then\n))) <<
-              %(do shell script "echo '#{key}'"\n)
+        statement << (((i += 1) == 1 ?
+          %(if response is "#{value}" then\n) :             %(else if response is "#{value}" then\n))) <<
```
Comment (ggilder) [inline lib/commander/user_interaction.rb]: Haha this code is almost too clever for its own good. Anyway looks like we can remove the extra spaces before `%(else ...`

File: lib/commander/user_interaction.rb [hunk 8]
```diff
@@ -310,15 +307,15 @@ def enable_paging
     #   end
     #
 
-    def progress arr, options = {}, &block
+    def progress(arr, options = {}, &_block)
```
Comment (ggilder) [inline lib/commander/user_interaction.rb]: Just delete unused block arg

File: lib/commander/user_interaction.rb [hunk 9]
```diff
@@ -428,50 +424,50 @@ def percent_complete
...
+          title: @title,
+          percent_complete: percent_complete,
+          progress_bar: progress_bar,
+          step: @step,
+          steps_remaining: steps_remaining,
+          total_steps: @total_steps,
+          time_elapsed: '%0.2fs' % time_elapsed,
+          time_remaining: @step > 0 ? '%0.2fs' % time_remaining : ''
+        }
+          .merge! @tokens
```
Comment (ggilder) [inline lib/commander/user_interaction.rb]: Join with the previous line

File: lib/commander/user_interaction.rb [hunk 10]
```diff
@@ -428,50 +424,50 @@ def percent_complete
...
-        }.
-        merge! @tokens
+          title: @title,
+          percent_complete: percent_complete,
+          progress_bar: progress_bar,
+          step: @step,
+          steps_remaining: steps_remaining,
+          total_steps: @total_steps,
+          time_elapsed: '%0.2fs' % time_elapsed,
+          time_remaining: @step > 0 ? '%0.2fs' % time_remaining : ''
```
Comment (ggilder) [inline lib/commander/user_interaction.rb]: I guess turning on the trailing commas option would add a lot more noise to this PR... maybe in a followup :wink:

File: .rubocop.yml [hunk 11]
```diff
@@ -0,0 +1,10 @@
+inherit_from: rubocop-todo.yml
+
+# Offense count: 5
```
Comment (ggilder) [inline .rubocop.yml]: Don't think we should have offense counts in files that aren't auto-generated

File: commander.gemspec [hunk 12]
```diff
@@ -1,25 +1,26 @@
...
-  s.add_runtime_dependency("highline", "~> 1.6.11")
+  s.add_runtime_dependency('highline', '~> 1.6.11')
 
-  s.add_development_dependency("rspec", "~> 2.14")
-  s.add_development_dependency("rake")
-  s.add_development_dependency("simplecov")
+  s.add_development_dependency('rspec', '~> 2.14')
+  s.add_development_dependency('rake')
+  s.add_development_dependency('simplecov')
+  s.add_development_dependency('rubocop')
```
Comment (ggilder) [inline commander.gemspec]: I think rubocop version should always be pinned because it changes frequently and so unrelated changes can break the build. While we're on the topic, it seems this was generated with an older version of rubocop - 0.20 vs current version 0.28. Can we put this on the current version?

File: spec/core_ext/object_spec.rb [hunk 13]
```diff
@@ -1,21 +1,19 @@
...
       expect(lambda {}.get_binding).to be_instance_of(Binding)
     end
   end
 
-  describe "#method_missing" do
-    it "should preserve its original behavior for missing methods" do
-      expect { i_am_a_missing_method() }.to raise_error(NoMethodError)
+  describe '#method_missing' do
+    it 'should preserve its original behavior for missing methods' do
+      expect { send('i_am_a_missing_method') }.to raise_error(NoMethodError)
```
Comment (ggilder) [inline spec/core_ext/object_spec.rb]: this could be symbol instead of string, yeah?

**Comment by ggilder** [comment]
This is awesome and long overdue! Thank you so much for taking the time to set this up. Left a few comments but this looks very close to being ready to merge.

File: Rakefile (hunk 1 above)
Comment (ralfclaassens) [inline Rakefile]: Agree, let's do that.

File: bin/commander (hunk 2 above)
Comment (ralfclaassens) [inline bin/commander]: Very strange, good comment. I'll correct this.

File: bin/commander (hunk 3 above)
Comment (ralfclaassens) [inline bin/commander]: Very strange, good comment. I'll correct this.

File: bin/commander (hunk 4 above)
Comment (ralfclaassens) [inline bin/commander]: It is probably a bug in Rubocop indeed. If you don't mind, I leave the test case in Rubocop for now...

**Comment by ggilder** [comment]
Can you rebase & squash to get rid of the "Merge branch 'master' into rubocop" commit?

**Comment by ggilder** [comment]
Moved to https://github.com/tj/commander/pull/92
